controllers/vdoconfig_controller.go

```diff
@@ -609,51 +604,13 @@ func (r *VDOConfigReconciler) updateVdoConfigWithNodeStatus(ctx context.Context,
 	return nil
 }
 
-func (r *VDOConfigReconciler) reconcileNodeTaint(ctx context.Context, clientset *kubernetes.Clientset) error {
-	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "Unable to fetch list of nodes")
-	}
-
-nodeLoop:
-	for _, node := range nodes.Items {
-		for _, taint := range node.Spec.Taints {
-			if taint.Key == CLOUD_PROVIDER_INIT_TAINT_KEY {
-				continue nodeLoop
-			}
-		}
-
-		r.Logger.Info("adding taint", "name", CLOUD_PROVIDER_INIT_TAINT_KEY)
-
-		taint := v1.Taint{
-			Key:       CLOUD_PROVIDER_INIT_TAINT_KEY,
-			Value:     "true",
-			Effect:    TAINT_NOSCHEDULE_KEY,
-			TimeAdded: nil,
-		}
-
-		node.Spec.Taints = append(node.Spec.Taints, taint)
-
-		_, err := clientset.CoreV1().Nodes().Update(ctx, &node, metav1.UpdateOptions{})
-		if err != nil {
-			return errors.Wrapf(err, "Error when updating taint %s on node %s", CLOUD_PROVIDER_INIT_TAINT_KEY, node.Name)
-		}
-	}
-
-	return nil
-}
-
-func (r *VDOConfigReconciler) reconcileNodeLabel(ctx vdocontext.VDOContext, req ctrl.Request, clientset *kubernetes.Clientset) error {
-	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "Unable to fetch list of nodes")
-	}
+func (r *VDOConfigReconciler) reconcileNodeLabel(ctx vdocontext.VDOContext, req ctrl.Request, clientset *kubernetes.Clientset, nodeList []v1.Node) error {
 
-	for _, node := range nodes.Items {
+	for _, node := range nodeList {
 		if _, ok := node.Labels[VDO_NODE_LABEL_KEY]; !ok {
```

Comment:
This can be re-written as 
```
if _, ok := node.Labels[VDO_NODE_LABEL_KEY]; ok {
       continue
}

r.Logger.Info("adding node label", "name", VDO_NODE_LABEL_KEY,"node",node.name)
node.Labels[VDO_NODE_LABEL_KEY] = req.Name
_, err := clientset.CoreV1().Nodes().Update(ctx, &node, metav1.UpdateOptions{})
if err != nil {
				return errors.Wrapf(err, "Unable to update label on node")
}
```